com.unity.netcode.gameobjects/Tests/Editor/StartStopTests.cs

```diff
@@ -0,0 +1,55 @@
+using NUnit.Framework;
+using UnityEngine;
+
+namespace Unity.Netcode.EditorTests
+{
+    public class StartStopTests
+    {
+        [Test]
+        public void TestStopAndRestartForExceptions()
```

Comment:
You're testing between 2 and (arguably) 12 things in this test.

- Are you trying to test IsServer, IsClient, IsHost are set properly for StartServer? If so, it should be its own test (asserts)
- Are you trying to test IsServer, IsClient, IsHost are set to false after Shutdown? If so, it should be its own test (asserts)
- Are you trying to test the flags are set properly on restart? If so, its own test (asserts)
- Are you trying to test multiple shutdowns don't throw an exception (as the test is named)? If so, should be its own test (asserts)
- Are you trying to ensure other values are properly set/cleared during a shutdown? Separate test... etc.

Reply:
Also, for this type of code, consider using a multi assert pattern (though be careful not to abuse the purpose of this)

https://docs.nunit.org/articles/nunit/writing-tests/assertions/multiple-asserts.html

Reply:
Thanks, will update!